### PR TITLE
Refactor `remote.app.get.Path()` to main process

### PR DESCRIPTION
**Pull request by tidy-dev**
## Description
This refactors the `remote` module usage in `remote.app.getPath(pathname)` to the main-process.

Impacts:
For the following, obtaining this path becomes asynchronous. 
- Clone Repository - is used to determine the default clone path
- Create Repository - is used to determine the default create path
- Creating Tutorial Repo - used to determine repo path

Note: This is one of several PR's to refactor our usage of the `remote` module to instead use the IPC messaging with the main process so that we can remove our [undesired](https://nornagon.medium.com/electrons-remote-module-considered-harmful-70d69500f31) `remote` module dependency (supporting ultimate goal of upgrading to electron 16).

## Release notes
Notes: no-notes

### CODE REVIEW

File: app/src/ui/lib/app-proxy.ts [hunk 1]
```diff
@@ -60,14 +79,13 @@ export function getUserDataPath(): string {
  *
  * This is preferable to using `remote` directly because we cache the result.
  */
-export function getDocumentsPath(): string {
+export async function getDocumentsPath(): Promise<string> {
   if (!documentsPath) {
-    const app = getApp()
```
Comment (sergiou87) [inline app/src/ui/lib/app-proxy.ts:52]: Now that #13803 is merged, I think you can also remove the `getApp` function entirely, right?

**Review by sergiou87 — COMMENTED** [review]
It looks and works great! Just a comment/suggestion that you might tackle in this PR.

File: app/src/ui/lib/app-proxy.ts (hunk 1 above)
Comment (tidy-dev) [reply app/src/ui/lib/app-proxy.ts:52]: Yep! That was my plan - not sure it would have even built after merging in dev.

**Review by sergiou87 — APPROVED** [review]
✨

**Pull request merged**